sql_export/tests/test_sql_query.py

```diff
@@ -0,0 +1,79 @@
+#    Author: Florian da Costa
+#    Copyright 2015 Akretion
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distnaributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from openerp.tests.common import TransactionCase
+import base64
+from openerp.osv.orm import except_orm
+
+
+class TestExportSqlQuery(TransactionCase):
+
+    def setUp(self):
+        super(TestExportSqlQuery, self).setUp()
+        query_vals = {
+            'name': 'test',
+            'query': "SELECT name, street FROM res_partner;"
+        }
+        self.sql_model = self.registry('sql.export')
+        self.query_id = self.sql_model.create(
+            self.cr,
+            self.uid,
+            query_vals)
+
+    def test_sql_query(self):
+        test = self.sql_model.export_sql_query(
+            self.cr, self.uid, [self.query_id])
+        wizard = self.registry('sql.file.wizard').browse(
+            self.cr, self.uid, test['res_id'])
+        export = base64.b64decode(wizard.file)
+        self.assertEqual(export.split(';')[0], 'name')
+        self.assertTrue(len(export.split(';')) > 6)
+
+    def test_prohibited_queries_creation(self):
+        prohibited_query1 = {
+            'name': "test update",
+            'query': """upDaTe sale_order SET partner_id = 1 WHERE id = 1"""
+        }
+        prohibited_query2 = {
+            'name': "test delete",
+            'query': """DELETE FROM sale_order;"""
+        }
+        prohibited_query3 = {
+            'name': "test delete2",
+            'query': """  DELETE FROM sale_order   ;"""
+        }
+        prohibited_query4 = {
+            'name': 'test delete3',
+            'query': """DELETE
+                        FROM
+                        sale_order
+            """
+        }
+        wrong_syntax_query = {
+            'name': 'test delete3',
+            'query': """SELCT id FROM sale_order"""
+        }
+        ok_query = {
+            'name': 'test ok',
+            'query': "SELECT create_date FROM sale_order"
+        }
+        with self.assertRaises(except_orm):
+            self.sql_model.create(self.cr, self.uid, prohibited_query1)
+            self.sql_model.create(self.cr, self.uid, prohibited_query2)
```

Comment:
You should repeat the 'with' statement before each call to `self.sql_model.create`
Currently, the assertion is true once one call fails. 
```python
def test_prohibited_queries_creation(self):
    prohibited_queries = [
        "upDaTe sale_order SET partner_id = 1 WHERE id = 1",
        "DELETE FROM sale_order;",
        "  DELETE FROM sale_order   ;",
        """DELETE
         FROM
         sale_order
         """,
    ]
    for query in prohibited_queries:
        with self.assertRaises(except_orm):
            self.sql_model.create(self.cr, self.uid, {'name': 'test_prohibited',
                                                      'query': query})
    wrong_syntax_query = {
        'name': 'test delete3',
        'query': """SELCT id FROM sale_order"""
    }
    with self.assertRaises(except_orm):
        self.sql_model.create(self.cr, self.uid, wrong_syntax_query)

    ok_query = {
       'name': 'test ok',
        'query': "SELECT create_date FROM sale_order"
    }
    query_id = self.sql_model.create(self.cr, self.uid, ok_query)
    self.assertIsNotNone(query_id)